### DISCUSSION TITLE
RAG on uploaded files / web search can only work for the first chat

### DISCUSSION THREAD

**Discussion by foxworld306** [open-webui/open-webui#4779] [General]
# Bug Report

## Installation Method

[Describe the method you used to install the project, e.g., git clone, Docker, pip, etc.]

## Environment

- **Open WebUI Version: 0.3.13
- **Ollama (if applicable): 0.3.5
- ** Web search tool: Searxng:latest

- **Operating System: Ubuntu 22.04
- **Browser (if applicable): Chrome, Edge

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [x ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ x] I have included the Docker container logs.
- [x ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

- Continuous chat on file content
- Using web search multiple times in a conversation session

## Actual Behavior:

I found that in the Open Webui environment I deployed, it is not possible to continuously have RAG conversations with uploaded documents. It only works for the first conversation in the New Chat session I opened. From the second conversation onwards, it seems that I have never uploaded any documents. But when I look at the logs of Open Webui, I can see that the backend has already provided document retrieval results. It seems that starting from the second conversation, the retrieval results were not provided to LLM as expected, only LLM's knowledge was used to answer.
This issue also affects my online search operations. I must perform the search action in the first conversation of each New Chat session, otherwise starting from the second conversation, even if I can see Searxing returning the search results, I won't be able to get the correct answer.

I have already tried:

- Place the document in workspace ->documents
- Using different LLM models (Qwen2 and GLM4)
- reset vector storage
- Using or not using Hybrid search,
- Different Embedding Models
- Different Rerank models
- Using different web search methods, including searxng, google pse, duckduckgo, tavily.
- Deploy different Open Webui (main / cuda)

All the above operations cannot solve this problem.

Can you please help to check?

## Logs and Screenshots
![image](https://github.com/user-attachments/assets/64128dcd-5e25-4a49-889c-299d412bbc64)
![image](https://github.com/user-attachments/assets/3b928152-c107-4aa2-a6ac-2532d3838e2b)
![image](https://github.com/user-attachments/assets/81fffbac-af4d-4047-ba2e-e0e9b2b9ccb2)

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by GridexX** [comment]
I have a related issue.
Open Web UI updated with the tag released 2 days ago `git-ec99ac7`.
Ollama at `latest`

The content of the file is not used.
tried with several documents. 
Here are some screenshots:


It's in french, but it means that it lakes information to answer my question:

![image](https://github.com/user-attachments/assets/00f8f1dc-8d0a-47d2-b1cd-b3ca4c3e1b99)

**Comment by raydoomed** [comment]
不仅是文档，他不能像游览器那样，每次都根据关键字生成新的内容，我也发现此问题，第一次搜索可以网络搜索并且得到正确答案，当第二次搜索时，他调用的是本地知识库，并不会进行网络搜索，哪怕强制网络搜索，也不会使用搜索结果中的内容去回答，始终用他的知识库数据

**Comment by feureau** [comment]
I’m experiencing the same problem described in this thread. Despite upgrading to **v0.6.13**, the RAG functionality with **uploaded documents** remains inconsistent:  

- **First document works** in a new chat session, but subsequent uploads or queries **fail to retrieve** from files.  
- This also affects the "**Paste Large Text as File**" feature—only the first paste is used despite being processed correctly.  
- Like others, I’ve noticed the backend retrieves results (visible in logs), but they’re **not used by the LLM** after the first interaction.  

### Questions:  
1. Is there any **update or fix** in progress for this behavior?  
2. Are there **workarounds** (e.g., config tweaks) to force document re-processing in the same session?  

I’ve tried resetting the vector store, switching models (e.g., Llama 3, Mistral), and different embedding methods, but the issue persists.  

**Logs/Environment:**  
- OpenWebUI: v0.6.13
- Ollama: latest  
- OS: Ubuntu 22.04  

Would appreciate any guidance—thanks for the hard work on this project!
